### PR TITLE
Node V8 debugger support

### PR DESCRIPTION
**Pull request by paulvanbrenk**
Check the version of the node binary and either force Webkit protocol
(VS 2017), or block debugging with Message (VS 2015).

### CODE REVIEW

File: Nodejs/Product/Nodejs/Nodejs.cs [hunk 1]
```diff
@@ -30,6 +31,21 @@ public sealed class Nodejs {
         private const string NodejsRegPath = "Software\\Node.js";
         private const string InstallPath = "InstallPath";
 
+        public static Version GetNodeVersion(string path)
+        {
+            if (string.IsNullOrEmpty(path)) {
+                throw new ArgumentException("\'path\' is not set", "path");
```
Comment (billti) [inline Nodejs/Product/Nodejs/Nodejs.cs]: Let's not crash if the path wasn't found. Just default to the old protocol.

File: Nodejs/Product/Nodejs/Project/NodejsProjectLauncher.cs [hunk 2]
```diff
@@ -68,18 +68,27 @@ internal class NodejsProjectLauncher : IProjectLauncher {
...
+
 #if !DEV15
+            if (chromeProtocolRequired) {
+                Nodejs.ShowNodeVersionNotSupported();
+                return VSConstants.S_OK;
+            }
+
             bool useWebKitDebugger = false;
 #else
-            bool useWebKitDebugger = NodejsPackage.Instance.GeneralOptionsPage.UseWebKitDebugger;
```
Comment (billti) [inline Nodejs/Product/Nodejs/Project/NodejsProjectLauncher.cs]: Is this option no longer used then? Did you remove it from the Tools / Options?

File: Nodejs/Product/Nodejs/Resources.resx [hunk 3]
```diff
@@ -620,4 +620,7 @@ Error retrieving websocket debug proxy information from web.config.</value>
   <data name="ErrorInvalidLaunchUrl" xml:space="preserve">
     <value>Launch url '{0}' is not valid </value>
   </data>
+  <data name="NodejsVersionNotSupported" xml:space="preserve">
+    <value>Your project is configured to use Node V8 or newer. Debugging Node V8 is not supported in VS 2015, please upgrade to VS 2017.</value>
```
Comment (paulvanbrenk) [inline Nodejs/Product/Nodejs/Resources.resx]: changed

File: Nodejs/Product/Nodejs/Nodejs.cs (hunk 1 above)
Comment (paulvanbrenk) [inline Nodejs/Product/Nodejs/Nodejs.cs]: agreed

**Comment by dlebu** [comment]
LGTM!

**Pull request merged**